doc/source/developing/extensions.rst

```diff
@@ -35,6 +35,32 @@ In subsequent versions, we plan to include in yt a catalog of known extensions
 and where to find them; this will put discoverability directly into the code
 base.
 
+Frontend as an extension
+------------------------
+
+Starting with version 4.2 of yt, any externally installed package that exports
+:class:`~yt.data_objects.static_output.Dataset` subclass as an entrypoint in
+``yt.frontends`` namespace in ``setup.py``:
+
+.. code-block:: python
+
+   setup(
+       # ...,
+       entry_points={"yt.frontends": ["myFrontend = my_frontend.api.MyFrontendDataset"]}
+   )
+
+or ``pyproject.toml``:
+
+.. code-block:: toml
+
+   [project.entry-points."yt.frontends"]
+   IdefixVtkDataset = "yt_idefix.data_structures:IdefixVtkDataset"
+   IdefixDmpDataset = "yt_idefix.data_structures:IdefixDmpDataset"
+   PlutoVtkDataset = "yt_idefix.data_structures:PlutoVtkDataset"
```

Comment:
love the easter egg, but I think the setup.py and pyproject.toml example should be equivalent (best to include multiple entry points in both)